### ISSUE TITLE
[AdManager][Banner] An ad is not rendered properly

### ISSUE THREAD

**Issue by yoalex5** [prebid/prebid-mobile-ios#175]
## Type of issue
Bug

## Description
When running a standard banner adunit it is not rendered by AdManager SDK. 

## Steps to reproduce
1 - Follow the [Guide to DFP setup](http://prebid.org/adops/step-by-step.html)
2 - Make a [Code Integration for iOS](http://prebid.org/prebid-mobile/pbm-api/ios/code-integration-ios.html)
3 - [Setup BannerAdUnit for Google Mobile Ads](http://prebid.org/prebid-mobile/pbm-api/ios/pbm-bannerad-ios.html)
4 - Run and check the result. When a banner ad is returned, the rendering fails.

## Test config
Host = Appnexus
ServerAccountId = "bfa84af2-bd16-4d35-96ad-31c6bb888df0"
configId = "6ace8c7d-88c0-4623-8117-75bc3f0a2e45"
adUnitID = /5300653/test_adunit_pavliuchyk_300x250_puc_ucTagData_prebid-server.rubiconproject.com

Charles log: [ios_Pv1.0_AMv7.42.2_prod_appnexus_rubicon-puc_ucTagData.chls.zip](https://github.com/prebid/prebid-mobile-ios/files/3038928/ios_Pv1.0_AMv7.42.2_prod_appnexus_rubicon-puc_ucTagData.chls.zip)

### Expected results
If AdManager response contains an ad it should be rendered

### Actual results
A creative is not showed

## Platform details
Prebid SDK v 1.0, git hash: beeee28699c268d021a7ad612ec6d60e64624f30
AdManager v 7.42.2

**Comment by AntoineJac** [comment]
I am confirming the issue, the GADBannerView has the correct size but the GADWebAdView is keeping the 1x1 size. The "x-afma-ad-size" header used by the SDK to build the GADWebAdView, this header field is returning 1x1 instead of the real creative size. The other header "X-Afma-Ad-Slot-Size" has the correct size.
It is an issue with the DFP SDK and I raised it to google a few week ago.

I have found a fix using the mraid.resize() in the prebid creative but hope Google could fix it...

**Comment by bszekely1** [comment]
To add additional details to the issue, Oleksandr who created this ticket has done some analysis on which versions of the Google ads SDK exhibit the failure to resize below:
<img width="241" alt="Screen Shot 2019-04-05 at 3 09 12 PM" src="https://user-images.githubusercontent.com/49168870/55659088-e4734480-57b4-11e9-85fd-aba57ec030f2.png">
<img width="544" alt="Screen Shot 2019-04-05 at 3 09 21 PM" src="https://user-images.githubusercontent.com/49168870/55659089-e4734480-57b4-11e9-98f3-0ef39c543995.png">

Oleksandr and I are recommending we add some logic in the universal creative (possibly with MRAID or pure JS) to resize the ad slot to the dimensions of the creative being served in SDK environments.

Is anyone else able to validate this to be an issue?

**Comment by bszekely1** [comment]
This issue was posted in a Google Forum support page here: [https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!category-topic/google-admob-ads-sdk/ios/648jzAP2EQY](url)

If this issue can be validated by others, I would like to get momentum behind the google support post to encourage Google to address the issue. If anyone is able to provide additional details, please post both here and the Google ticket.

**Comment by yoalex5** [comment]
As a variant of solution that should be implemented from a publisher side it is using
`DFPBannerView.resize` function in `GADBannerViewDelegate.adViewDidReceiveAd` 

For example

```
func adViewDidReceiveAd(_ bannerView: GADBannerView) {
    print("adViewDidReceiveAd")

    self.dfpBanner.resize(bannerView.adSize)
}
```

**Comment by ppuviarasu** [comment]
This needs to be documented here http://prebid.org/prebid-mobile/pbm-api/ios/pbm-bannerad-ios.html && http://prebid.org/prebid-mobile/pbm-api/android/banneradunit-android.html to include adManager and the implementation logics like these to address specific mediation partner issues

**Comment by AntoineJac** [comment]
I have run some tests on iOS and Android for different SDK version and seems to work well.
This solution seems to be the simplest and easy to setup. 
Methods used to resize are officially supported and part of Google api doc so it is also safe to go but will need to be removed when Google confirm the issue is fixed!

**Comment by bszekely1** [comment]
Released in 1.1.1
